doc/source/ray-contribute/development.rst

```diff
@@ -143,6 +143,8 @@ To build Ray on Ubuntu, run the following commands:
   nvm install 14
   nvm use 14
 
+.. note::
+  We recommend using `bazelisk` for building. If you prefer to use `bazel`, ensure it is version `5.4.1`.
```

Comment:
Does it need to be exactly `5.4.1`? Can you explain a bit more about this (either in the note or in the PR description)? Not enough context right now to understand why it needs to be `5.4.1`.

Reply:
yes, and it is pinned her:
https://github.com/anyscale/product/blob/master/.bazelversion

different bazel version has different options / flags / skylark grammar support.

all versions should be pinned.

Reply:
the doc here probably should just say use `bazelisk`, or use the same version as `.bazelversion` specifies.

behavior of building with other bazel versions is undefined.

Reply:
Updated the doc. Feel free to make some change if it's not accurate.